subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -157,21 +251,77 @@ u32_t ll_wl_remove(bt_addr_le_t *addr)
 		return 0;
 	}
 
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	return wl_peers_remove(addr);
+#else
 	return filter_remove(&wl, addr->type, addr->a.val);
+#endif /* CONFIG_BLUETOOTH_CONTROLLER_PRIVACY */
 }
 
 #if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
 
-#define RL_MATCH(i, id_addr_type, id_addr) (rl[i].taken && \
-		    rl[i].peer_id_addr_type == (id_addr_type & 0x1) && \
-		    !memcmp(rl[i].peer_id_addr.val, id_addr, BDADDR_SIZE))
+static void filter_wl_update(void)
+{
+	int i;
+	/* Populate filter from wl peers */
+	filter_clear(&wl);
 
+	for (i = 0; i < WL_SIZE; i++) {
+		int j = wl_peers[i].rl_idx;
+		if (!rl_enable || j == IDX_NONE || !rl[j].pirk || rl[j].dev) {
```

Comment:
if you have the check for rl_enable outside the loop, you just need to loop i times calling filter_insert without carrying for reading j. This is just a optimization, more code, but faster by few cycles for each address.